src/segments/nba.go

```diff
@@ -0,0 +1,511 @@
+	BroadcasterName   *string       `json:"broadcasterName,omitempty"`
+	TapeDelayComments *string       `json:"tapeDelayComments,omitempty"`
+}
+
+type Broadcaster struct {
+	BroadcastID       string `json:"broadcastID"`
+	BroadcasterName   string `json:"broadcasterName"`
+	TapeDelayComments string `json:"tapeDelayComments"`
+}
+
+func (nba *NBA) Template() string {
+	switch nba.GameStatus {
+	case Scheduled:
+		return nba.scheduledTemplate()
+	case InProgress:
+		return nba.inProgressTemplate()
+	case Finished:
+		return nba.finishedTemplate()
+	case NotFound:
+		return nba.notFoundTemplate()
+	default:
+		return nba.errorTemplate()
+	}
+}
+
+// Template to return if the game is scheduled
+func (nba *NBA) scheduledTemplate() string {
+	// At this point in time, the schedule endpoint does not have the team records
+	if nba.HomeTeamWins == 0 && nba.HomeTeamLosses == 0 && nba.AwayTeamWins == 0 && nba.AwayTeamLosses == 0 {
+		return " 󰠆 {{ .HomeTeam}} vs {{ .AwayTeam}} | {{.GameDate}} | {{.Time}} "
+	}
+	// On the day of the game, we may retreive 'Scheduled' games from the score endpoint which DOES contain the team records
+	return " 󰠆 {{ .HomeTeam}} ({{.HomeTeamWins}}-{{.HomeTeamLosses}}) vs {{ .AwayTeam}} ({{.AwayTeamWins}}-{{.AwayTeamLosses}}) | {{.GameDate}} | {{.Time}} "
+}
+
+// Template to return if the game is in progress
+func (nba *NBA) inProgressTemplate() string {
+	return " 󰠆 {{ .HomeTeam}} ({{.HomeTeamWins}}-{{.HomeTeamLosses}}):{{.HomeScore}} | {{.AwayTeam}} ({{.AwayTeamWins}}-{{.AwayTeamLosses}}):{{ .AwayScore}} | {{.Time}} "
+}
+
+// Template to return if the game is finished
+func (nba *NBA) finishedTemplate() string {
+	return " 󰠆 {{ .HomeTeam}} ({{.HomeTeamWins}}-{{.HomeTeamLosses}}):{{.HomeScore}} | {{.AwayTeam}} ({{.AwayTeamWins}}-{{.AwayTeamLosses}}):{{ .AwayScore}} | {{.Time}} "
+}
+
+// Template to return if the game is not found
+func (nba *NBA) notFoundTemplate() string {
+	daysOffset := nba.props.GetInt(DaysOffset, 8)
+	return fmt.Sprintf(" 󰠆 No game found for team within the next %d days", daysOffset)
+}
+
+// Template to return if the game is in an unhandled state
+func (nba *NBA) errorTemplate() string {
+	return " 󰠆 There was an error in retrieving the data"
+}
+
+func (nba *NBA) Enabled() bool {
+	data, err := nba.getResult()
+	if err != nil {
+		return false
+	}
+
+	nba.NBAData = *data
+
+	return true
+}
+
+// Returns an empty Game Data struct with the GameStatus set to NotFound
+// Helpful for caching the fact that a game was not found for a team
+func (nba *NBA) getGameNotFoundData() string {
+	gameNotFound := NBAData{
+		HomeTeam:       "",
+		AwayTeam:       "",
+		Time:           "",
+		GameDate:       "",
+		StartTimeUTC:   "",
+		GameStatus:     NotFound,
+		HomeScore:      0,
+		AwayScore:      0,
+		HomeTeamWins:   0,
+		HomeTeamLosses: 0,
+		AwayTeamWins:   0,
+		AwayTeamLosses: 0,
+	}
+	// get string representation of res, nba data
+	res, _ := json.Marshal(gameNotFound)
+	return string(res)
```

Comment:
this can be a string constant instead of a runtime parsing of an empty struct, no?

Reply:
True! It felt nice to always cache the same type of object and to take advantage of the GameStatus struct, but sounds like it may not be worth the extra performance hit?